foxes/models/wake_models/ti/iec_ti.py

```diff
@@ -0,0 +1,199 @@
+        mdata : foxes.core.Data
+            The model data
+        fdata : foxes.core.Data
+            The farm data
+        states_source_turbine : numpy.ndarray
+            For each state, one turbine index for the
+            wake causing turbine. Shape: (n_states,)
+        x : numpy.ndarray
+            The x values, shape: (n_states, n_points)
+        r : numpy.ndarray
+            The radial values for each x value, shape:
+            (n_states, n_points, n_r_per_x, 2)
+        ct : numpy.ndarray
+            The ct values of the wake-causing turbines,
+            shape: (n_states, n_points)
+
+        Returns
+        -------
+        wake_r : numpy.ndarray
+            The wake radii, shape: (n_states, n_points)
+
+        """
+        # prepare:
+        n_states = fdata.n_states
+        st_sel = (np.arange(n_states), states_source_turbine)
+
+        # get k:
+        k = self.get_data(FV.K, fdata, st_sel)
+        if isinstance(k, np.ndarray):
+            k = k[:, None]
+
+        # calculate:
+        radius = k * x
+
+        return radius
+
+    def calc_centreline_wake_deltas(
+        self, algo, mdata, fdata, states_source_turbine, sp_sel, x, wake_r, ct
+    ):
+        """
+        Calculate centre line results of wake deltas.
+
+        Parameters
+        ----------
+        algo : foxes.core.Algorithm
+            The calculation algorithm
+        mdata : foxes.core.Data
+            The model data
+        fdata : foxes.core.Data
+            The farm data
+        states_source_turbine : numpy.ndarray
+            For each state, one turbine index for the
+            wake causing turbine. Shape: (n_states,)
+        sp_sel : numpy.ndarray of bool
+            The state-point selection, for which the wake
+            is non-zero, shape: (n_states, n_points)
+        x : numpy.ndarray
+            The x values, shape: (n_sp_sel,)
+        wake_r : numpy.ndarray
+            The wake radii, shape: (n_sp_sel,)
+        ct : numpy.ndarray
+            The ct values of the wake-causing turbines,
+            shape: (n_sp_sel,)
+
+        Returns
+        -------
+        cl_del : dict
+            The centre line wake deltas. Key: variable name str,
+            varlue: numpy.ndarray, shape: (n_sp_sel,)
+
+        """
+        # prepare:
+        n_states = fdata.n_states
+        n_points = sp_sel.shape[1]
+        st_sel = (np.arange(n_states), states_source_turbine)
+
+        # read D from extra data:
+        D = np.zeros((n_states, n_points), dtype=FC.DTYPE)
+        D[:] = fdata[FV.D][st_sel][:, None]
+        D = D[sp_sel]
+
+
+        # get ws:
+        ws = np.zeros((n_states, n_points), dtype=FC.DTYPE)
+        ws[:] = fdata[FV.REWS][st_sel][:, None]
+        ws = ws[sp_sel]
+        
+        # get ct:
+        '''ct = np.zeros((n_states, n_points), dtype=FC.DTYPE)
+        ct[:] = fdata[FV.CT][st_sel][:, None]
+        ct = ct[sp_sel]'''
+        ct[ct > self.ct_max] = self.ct_max
```

Comment:
This is only relevant for 2019, right? Could you please move it there? Thanks!

Reply:
Ah I saw that it is already limited in the class `TopHatWakeModel`. So I don't need it here and I will remove it